### ISSUE TITLE
RFC: Splitting up Util.h

### ISSUE THREAD

**Issue by mmcgr** [souffle-lang/souffle#1437]
`Util.h` is a bit of a bloated monstrosity now. It includes a large selection of system headers and a very wide range of functionality. Splitting it up would make it more readable, and possibly improve compilation time.

I was thinking of this split:
```
CacheUtil.h
ContainerUtil.h
FileUtil.h
FunctionalUtil.h
MiscUtil.h
ParallelUtil.h
StreamUtil.h
StringUtil.h
```

It would also be nice to split utilities used for compiling souffle from evaluating souffle, but there's a lot of overlap. I've already done the split as above, but more input would be helpful so we don't face a similar problem in the future.

**Comment by mmcgr** [comment]
And now `EvaluatorUtil.h` for the evaluation specific utils.

**Comment by b-scholz** [comment]
I think it is time to split up the Util files; they become too bloated.

**Comment by b-scholz** [comment]
Also, we need to double-check whether all functions in the files are still in use - if not; we need to remove them.

**Comment by b-scholz** [comment]
Fixed by #1444.
